{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_value</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AssertionError('0 != 99')]]</td>\n",
       "      <td>283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IndexError('list index out of range')]]</td>\n",
       "      <td>257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IndexError('string index out of range')]]</td>\n",
       "      <td>230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AssertionError(\"'odde' != 'des'\\n- odde\\n+ de...</td>\n",
       "      <td>146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AssertionError(\"'des' != 'odde'\\n- des\\n+ odd...</td>\n",
       "      <td>145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TypeError(\"'&gt;' not supported between instance...</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>AssertionError('None != 0')]]</td>\n",
       "      <td>103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AttributeError(\"'int' object has no attribute...</td>\n",
       "      <td>103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AssertionError('59 != 0')]]</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AssertionError('1 != 2')]]</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>AssertionError('None != 1')]]</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>AssertionError('None != []')]]</td>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>NameError(\"name 'vare' is not defined\")]]</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>AssertionError(\"'1' != 1\")]]</td>\n",
       "      <td>62</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>TypeError(\"unsupported operand type(s) for -:...</td>\n",
       "      <td>52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TypeError(\"'int' object is not iterable\")]]</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>AssertionError(\"'0' != 99\")]]</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>TypeError(\"'int' object is not subscriptable\")]]</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>AssertionError(\"'odd' != 'odde'\\n- odd\\n+ odd...</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>AssertionError(\"'u' != 'a'\\n- u\\n+ a\\n\")]]</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>AssertionError('None != 2')]]</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>TypeError('can only concatenate str (not \"int...</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>AssertionError(\"None != 'odde'\")]]</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>TypeError(\"'&lt;=' not supported between instanc...</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         unique_value  count\n",
       "0                         AssertionError('0 != 99')]]    283\n",
       "1             IndexError('list index out of range')]]    257\n",
       "2           IndexError('string index out of range')]]    230\n",
       "3    AssertionError(\"'odde' != 'des'\\n- odde\\n+ de...    146\n",
       "4    AssertionError(\"'des' != 'odde'\\n- des\\n+ odd...    145\n",
       "5    TypeError(\"'>' not supported between instance...    108\n",
       "6                       AssertionError('None != 0')]]    103\n",
       "7    AttributeError(\"'int' object has no attribute...    103\n",
       "8                         AssertionError('59 != 0')]]     85\n",
       "9                          AssertionError('1 != 2')]]     80\n",
       "10                      AssertionError('None != 1')]]     80\n",
       "11                     AssertionError('None != []')]]     71\n",
       "12          NameError(\"name 'vare' is not defined\")]]     69\n",
       "13                       AssertionError(\"'1' != 1\")]]     62\n",
       "14   TypeError(\"unsupported operand type(s) for -:...     52\n",
       "15        TypeError(\"'int' object is not iterable\")]]     42\n",
       "16                      AssertionError(\"'0' != 99\")]]     33\n",
       "17   TypeError(\"'int' object is not subscriptable\")]]     33\n",
       "18   AssertionError(\"'odd' != 'odde'\\n- odd\\n+ odd...     33\n",
       "19         AssertionError(\"'u' != 'a'\\n- u\\n+ a\\n\")]]     29\n",
       "20                      AssertionError('None != 2')]]     28\n",
       "21   TypeError('can only concatenate str (not \"int...     26\n",
       "22                 AssertionError(\"None != 'odde'\")]]     25\n",
       "23   TypeError(\"'<=' not supported between instanc...     24"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "# Load the CSV file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H21/errorH21.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Drop the first column\n",
    "data = data.drop(data.columns[0], axis=1)\n",
    "\n",
    "#Get unique data and their count\n",
    "unique_values = data.stack().value_counts()\n",
    "\n",
    "unique_values_df = unique_values.rename_axis('unique_value').reset_index(name='count')\n",
    "\n",
    "# Display the grouped data as a table\n",
    "unique_values_df.head(24)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sum of NameError:  359\n",
      "Sum of AssertionError:  1712\n",
      "Sum of TypeError:  452\n",
      "Sum of IndexError:  499\n",
      "Sum of AttributeError:  151\n",
      "Sum of UnboundLocalError:  33\n",
      "Sum of ValueError:  62\n",
      "Sum of KeyError:  6\n"
     ]
    }
   ],
   "source": [
    "# Load the CSV file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H21/errorH21.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Drop the first column\n",
    "data = data.drop(data.columns[0], axis=1)\n",
    "\n",
    "#Get unique data and their count\n",
    "unique_values = data.stack().value_counts()\n",
    "\n",
    "# Filter unique values containing \"NameError\"\n",
    "name_error_values = [value for value in unique_values.index if \"NameError\" in value]\n",
    "name_error_counts = unique_values[name_error_values]\n",
    "\n",
    "# Sum the counts of the NameError values\n",
    "name_error_sum = name_error_counts.sum()\n",
    "\n",
    "# Filter unique values containing \"AssertionError\"\n",
    "assertion_error_values = [value for value in unique_values.index if \"AssertionError\" in value]\n",
    "assertion_error_counts = unique_values[assertion_error_values]\n",
    "\n",
    "# Sum the counts of the AssertionError values\n",
    "assertion_error_sum = assertion_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"TypeError\"\n",
    "type_error_values = [value for value in unique_values.index if \"TypeError\" in value]\n",
    "type_error_counts = unique_values[type_error_values]\n",
    "\n",
    "# Sum the counts of the TypeError values\n",
    "type_error_sum = type_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"IndexError\"\n",
    "index_error_values = [value for value in unique_values.index if \"IndexError\" in value]\n",
    "index_error_counts = unique_values[index_error_values]\n",
    "\n",
    "# Sum the counts of the IndexError values\n",
    "index_error_sum = index_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"AttributeError\"\n",
    "attribute_error_values = [value for value in unique_values.index if \"AttributeError\" in value]\n",
    "attribute_error_counts = unique_values[attribute_error_values]\n",
    "\n",
    "# Sum the counts of the AttributeError values\n",
    "attribute_error_sum = attribute_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"UnboundLocalError\"\n",
    "unbound_local_error_values = [value for value in unique_values.index if \"UnboundLocalError\" in value]\n",
    "unbound_local_error_counts = unique_values[unbound_local_error_values]\n",
    "\n",
    "# Sum the counts of the UnboundLocalError values\n",
    "unbound_local_error_sum = unbound_local_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"ValueError\"\n",
    "value_error_values = [value for value in unique_values.index if \"ValueError\" in value]\n",
    "value_error_counts = unique_values[value_error_values]\n",
    "\n",
    "# Sum the counts of the ValueError values\n",
    "value_error_sum = value_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"KeyError\"\n",
    "key_error_values = [value for value in unique_values.index if \"KeyError\" in value]\n",
    "key_error_counts = unique_values[key_error_values]\n",
    "\n",
    "# Sum the counts of the KeyError values\n",
    "key_error_sum = key_error_counts.sum()\n",
    "\n",
    "# Display the grouped data as a table\n",
    "print(\"Sum of NameError: \", name_error_sum)\n",
    "print(\"Sum of AssertionError: \", assertion_error_sum)\n",
    "print(\"Sum of TypeError: \", type_error_sum)\n",
    "print(\"Sum of IndexError: \", index_error_sum)\n",
    "print(\"Sum of AttributeError: \", attribute_error_sum)\n",
    "print(\"Sum of UnboundLocalError: \", unbound_local_error_sum)\n",
    "print(\"Sum of ValueError: \", value_error_sum)\n",
    "print(\"Sum of KeyError: \", key_error_sum)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_value</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AssertionError('False is not true : False')]]</td>\n",
       "      <td>554</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IndexError('list index out of range')]]</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AssertionError('False is not true : True')]]</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NameError(\"name 'lenght' is not defined\")]]</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NameError(\"name 'np' is not defined\")]]</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>TypeError(\"'int' object is not iterable\")]]</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ValueError('The truth value of an array with ...</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>IndexError('index 4 is out of bounds for axis...</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>IndexError('arrays used as indices must be of...</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AssertionError('None is not true : False')]]</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>AssertionError('None is not true : True')]]</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>IndexError('index 5 is out of bounds for axis...</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>AssertionError('0 != 11')]]</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>AssertionError('3 != 11')]]</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>IndexError('pop index out of range')]]</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>NameError(\"name 'length' is not defined\")]]</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>IndexError('index 4 is out of bounds for axis...</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>AssertionError('0 is not true : 3')]]</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>AssertionError('None is not true : 3')]]</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>AssertionError('4 != 11')]]</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>AssertionError('None != 11')]]</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>AssertionError('7 != 11')]]</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>AssertionError('6 != 11')]]</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>TypeError(\"'NoneType' object is not iterable\")]]</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         unique_value  count\n",
       "0       AssertionError('False is not true : False')]]    554\n",
       "1             IndexError('list index out of range')]]     50\n",
       "2        AssertionError('False is not true : True')]]     47\n",
       "3         NameError(\"name 'lenght' is not defined\")]]     43\n",
       "4             NameError(\"name 'np' is not defined\")]]     41\n",
       "5         TypeError(\"'int' object is not iterable\")]]     38\n",
       "6    ValueError('The truth value of an array with ...     34\n",
       "7    IndexError('index 4 is out of bounds for axis...     33\n",
       "8    IndexError('arrays used as indices must be of...     33\n",
       "9        AssertionError('None is not true : False')]]     30\n",
       "10        AssertionError('None is not true : True')]]     24\n",
       "11   IndexError('index 5 is out of bounds for axis...     22\n",
       "12                        AssertionError('0 != 11')]]     20\n",
       "13                        AssertionError('3 != 11')]]     19\n",
       "14             IndexError('pop index out of range')]]     19\n",
       "15        NameError(\"name 'length' is not defined\")]]     17\n",
       "16   IndexError('index 4 is out of bounds for axis...     17\n",
       "17              AssertionError('0 is not true : 3')]]     17\n",
       "18           AssertionError('None is not true : 3')]]     16\n",
       "19                        AssertionError('4 != 11')]]     14\n",
       "20                     AssertionError('None != 11')]]     12\n",
       "21                        AssertionError('7 != 11')]]     12\n",
       "22                        AssertionError('6 != 11')]]     11\n",
       "23   TypeError(\"'NoneType' object is not iterable\")]]     11"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "# Load the CSV file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H22/errorH22morgen.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Drop the first column\n",
    "data = data.drop(data.columns[0], axis=1)\n",
    "\n",
    "#Get unique data and their count\n",
    "unique_values = data.stack().value_counts()\n",
    "\n",
    "unique_values_df = unique_values.rename_axis('unique_value').reset_index(name='count')\n",
    "\n",
    "# Display the grouped data as a table\n",
    "unique_values_df.head(24)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sum of NameError:  240\n",
      "Sum of AssertionError:  814\n",
      "Sum of TypeError:  149\n",
      "Sum of IndexError:  181\n",
      "Sum of AttributeError:  41\n",
      "Sum of UnboundLocalError:  31\n",
      "Sum of ValueError:  46\n",
      "Sum of KeyError:  0\n"
     ]
    }
   ],
   "source": [
    "# Load the CSV file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H22/errorH22morgen.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Drop the first column\n",
    "data = data.drop(data.columns[0], axis=1)\n",
    "\n",
    "#Get unique data and their count\n",
    "unique_values = data.stack().value_counts()\n",
    "\n",
    "# Filter unique values containing \"NameError\"\n",
    "name_error_values = [value for value in unique_values.index if \"NameError\" in value]\n",
    "name_error_counts = unique_values[name_error_values]\n",
    "\n",
    "# Sum the counts of the NameError values\n",
    "name_error_sum = name_error_counts.sum()\n",
    "\n",
    "# Filter unique values containing \"AssertionError\"\n",
    "assertion_error_values = [value for value in unique_values.index if \"AssertionError\" in value]\n",
    "assertion_error_counts = unique_values[assertion_error_values]\n",
    "\n",
    "# Sum the counts of the AssertionError values\n",
    "assertion_error_sum = assertion_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"TypeError\"\n",
    "type_error_values = [value for value in unique_values.index if \"TypeError\" in value]\n",
    "type_error_counts = unique_values[type_error_values]\n",
    "\n",
    "# Sum the counts of the TypeError values\n",
    "type_error_sum = type_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"IndexError\"\n",
    "index_error_values = [value for value in unique_values.index if \"IndexError\" in value]\n",
    "index_error_counts = unique_values[index_error_values]\n",
    "\n",
    "# Sum the counts of the IndexError values\n",
    "index_error_sum = index_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"AttributeError\"\n",
    "attribute_error_values = [value for value in unique_values.index if \"AttributeError\" in value]\n",
    "attribute_error_counts = unique_values[attribute_error_values]\n",
    "\n",
    "# Sum the counts of the AttributeError values\n",
    "attribute_error_sum = attribute_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"UnboundLocalError\"\n",
    "unbound_local_error_values = [value for value in unique_values.index if \"UnboundLocalError\" in value]\n",
    "unbound_local_error_counts = unique_values[unbound_local_error_values]\n",
    "\n",
    "# Sum the counts of the UnboundLocalError values\n",
    "unbound_local_error_sum = unbound_local_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"ValueError\"\n",
    "value_error_values = [value for value in unique_values.index if \"ValueError\" in value]\n",
    "value_error_counts = unique_values[value_error_values]\n",
    "\n",
    "# Sum the counts of the ValueError values\n",
    "value_error_sum = value_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"KeyError\"\n",
    "key_error_values = [value for value in unique_values.index if \"KeyError\" in value]\n",
    "key_error_counts = unique_values[key_error_values]\n",
    "\n",
    "# Sum the counts of the KeyError values\n",
    "key_error_sum = key_error_counts.sum()\n",
    "\n",
    "# Display the grouped data as a table\n",
    "print(\"Sum of NameError: \", name_error_sum)\n",
    "print(\"Sum of AssertionError: \", assertion_error_sum)\n",
    "print(\"Sum of TypeError: \", type_error_sum)\n",
    "print(\"Sum of IndexError: \", index_error_sum)\n",
    "print(\"Sum of AttributeError: \", attribute_error_sum)\n",
    "print(\"Sum of UnboundLocalError: \", unbound_local_error_sum)\n",
    "print(\"Sum of ValueError: \", value_error_sum)\n",
    "print(\"Sum of KeyError: \", key_error_sum)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>unique_value</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IndexError('arrays used as indices must be of...</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AssertionError('0 != 15')]]</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IndexError('list index out of range')]]</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TypeError('only integer scalar arrays can be ...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ValueError('The truth value of an array with ...</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AssertionError('None is not true : M3')]]</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>TypeError(\"'int' object is not subscriptable\")]]</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>TypeError(\"'int' object is not iterable\")]]</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>IndexError('index 4 is out of bounds for axis...</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>NameError(\"name 'M3' is not defined\")]]</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>AssertionError('0 is not true : 38.2')]]</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>TypeError('only size-1 arrays can be converte...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>IndexError('index 4 is out of bounds for axis...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>TypeError(\"object of type 'int' has no len()\")]]</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>NameError(\"name 'i' is not defined\")]]</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>TypeError(\"'tuple' object is not callable\")]]</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>NameError(\"name 'true' is not defined\")]]</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>AssertionError('None is not true : 38.2')]]</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>AssertionError('None != 15')]]</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>AttributeError(\"'numpy.float64' object has no...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>UnboundLocalError(\"local variable 'sum' refer...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>TypeError(\"unhashable type: 'list'\")]]</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>TimeoutError()]]</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>AssertionError('8 != 15')]]</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         unique_value  count\n",
       "0    IndexError('arrays used as indices must be of...     11\n",
       "1                         AssertionError('0 != 15')]]     10\n",
       "2             IndexError('list index out of range')]]      9\n",
       "3    TypeError('only integer scalar arrays can be ...      7\n",
       "4    ValueError('The truth value of an array with ...      6\n",
       "5           AssertionError('None is not true : M3')]]      6\n",
       "6    TypeError(\"'int' object is not subscriptable\")]]      6\n",
       "7         TypeError(\"'int' object is not iterable\")]]      6\n",
       "8    IndexError('index 4 is out of bounds for axis...      5\n",
       "9             NameError(\"name 'M3' is not defined\")]]      4\n",
       "10           AssertionError('0 is not true : 38.2')]]      4\n",
       "11   TypeError('only size-1 arrays can be converte...      4\n",
       "12   IndexError('index 4 is out of bounds for axis...      4\n",
       "13   TypeError(\"object of type 'int' has no len()\")]]      4\n",
       "14             NameError(\"name 'i' is not defined\")]]      4\n",
       "15      TypeError(\"'tuple' object is not callable\")]]      3\n",
       "16          NameError(\"name 'true' is not defined\")]]      3\n",
       "17        AssertionError('None is not true : 38.2')]]      3\n",
       "18                     AssertionError('None != 15')]]      3\n",
       "19   AttributeError(\"'numpy.float64' object has no...      2\n",
       "20   UnboundLocalError(\"local variable 'sum' refer...      2\n",
       "21             TypeError(\"unhashable type: 'list'\")]]      2\n",
       "22                                   TimeoutError()]]      2\n",
       "23                        AssertionError('8 != 15')]]      2"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "# Load the CSV file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H22Kveld/errorH22kveld.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Drop the first column\n",
    "data = data.drop(data.columns[0], axis=1)\n",
    "\n",
    "#Get unique data and their count\n",
    "unique_values = data.stack().value_counts()\n",
    "\n",
    "unique_values_df = unique_values.rename_axis('unique_value').reset_index(name='count')\n",
    "\n",
    "# Display the grouped data as a table\n",
    "unique_values_df.head(24)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sum of NameError:  38\n",
      "Sum of AssertionError:  36\n",
      "Sum of TypeError:  44\n",
      "Sum of IndexError:  31\n",
      "Sum of AttributeError:  10\n",
      "Sum of UnboundLocalError:  8\n",
      "Sum of ValueError:  7\n",
      "Sum of KeyError:  0\n"
     ]
    }
   ],
   "source": [
    "# Load the CSV file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H22Kveld/errorH22kveld.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Drop the first column\n",
    "data = data.drop(data.columns[0], axis=1)\n",
    "\n",
    "#Get unique data and their count\n",
    "unique_values = data.stack().value_counts()\n",
    "\n",
    "# Filter unique values containing \"NameError\"\n",
    "name_error_values = [value for value in unique_values.index if \"NameError\" in value]\n",
    "name_error_counts = unique_values[name_error_values]\n",
    "\n",
    "# Sum the counts of the NameError values\n",
    "name_error_sum = name_error_counts.sum()\n",
    "\n",
    "# Filter unique values containing \"AssertionError\"\n",
    "assertion_error_values = [value for value in unique_values.index if \"AssertionError\" in value]\n",
    "assertion_error_counts = unique_values[assertion_error_values]\n",
    "\n",
    "# Sum the counts of the AssertionError values\n",
    "assertion_error_sum = assertion_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"TypeError\"\n",
    "type_error_values = [value for value in unique_values.index if \"TypeError\" in value]\n",
    "type_error_counts = unique_values[type_error_values]\n",
    "\n",
    "# Sum the counts of the TypeError values\n",
    "type_error_sum = type_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"IndexError\"\n",
    "index_error_values = [value for value in unique_values.index if \"IndexError\" in value]\n",
    "index_error_counts = unique_values[index_error_values]\n",
    "\n",
    "# Sum the counts of the IndexError values\n",
    "index_error_sum = index_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"AttributeError\"\n",
    "attribute_error_values = [value for value in unique_values.index if \"AttributeError\" in value]\n",
    "attribute_error_counts = unique_values[attribute_error_values]\n",
    "\n",
    "# Sum the counts of the AttributeError values\n",
    "attribute_error_sum = attribute_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"UnboundLocalError\"\n",
    "unbound_local_error_values = [value for value in unique_values.index if \"UnboundLocalError\" in value]\n",
    "unbound_local_error_counts = unique_values[unbound_local_error_values]\n",
    "\n",
    "# Sum the counts of the UnboundLocalError values\n",
    "unbound_local_error_sum = unbound_local_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"ValueError\"\n",
    "value_error_values = [value for value in unique_values.index if \"ValueError\" in value]\n",
    "value_error_counts = unique_values[value_error_values]\n",
    "\n",
    "# Sum the counts of the ValueError values\n",
    "value_error_sum = value_error_counts.sum()\n",
    "\n",
    "#Filter unique values containing \"KeyError\"\n",
    "key_error_values = [value for value in unique_values.index if \"KeyError\" in value]\n",
    "key_error_counts = unique_values[key_error_values]\n",
    "\n",
    "# Sum the counts of the KeyError values\n",
    "key_error_sum = key_error_counts.sum()\n",
    "\n",
    "# Display the grouped data as a table\n",
    "print(\"Sum of NameError: \", name_error_sum)\n",
    "print(\"Sum of AssertionError: \", assertion_error_sum)\n",
    "print(\"Sum of TypeError: \", type_error_sum)\n",
    "print(\"Sum of IndexError: \", index_error_sum)\n",
    "print(\"Sum of AttributeError: \", attribute_error_sum)\n",
    "print(\"Sum of UnboundLocalError: \", unbound_local_error_sum)\n",
    "print(\"Sum of ValueError: \", value_error_sum)\n",
    "print(\"Sum of KeyError: \", key_error_sum)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Oppgave 2a:  269\n",
      "Oppgave 2b:  131\n",
      "Oppgave 2c:  639\n",
      "Oppgave 2g:  310\n",
      "Oppgave 2h:  637\n",
      "Oppgave 2i:  514\n",
      "Oppgave 3.1:  774\n",
      "Oppgave 3.2:  15\n",
      "Oppgave 3.3:  0\n",
      "Oppgave 3.4:  0\n",
      "Oppgave 3.5:  0\n"
     ]
    }
   ],
   "source": [
    "# Load csv file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H21/errorH21.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Group the data into values that contain \"oppgave2a\"\n",
    "oppgave2a = data.stack().value_counts().filter(like='oppgave2a', axis=0)\n",
    "\n",
    "#Group the data into values that contain \"oppgave2b\"\n",
    "oppgave2b = data.stack().value_counts().filter(like='oppgave2b', axis=0)\n",
    "\n",
    "#Group the data into values that contain \"oppgave2c\"\n",
    "oppgave2c = data.stack().value_counts().filter(like='oppgave2c', axis=0)\n",
    "\n",
    "#Group the data into values that contain \"oppgave2g\"\n",
    "oppgave2g = data.stack().value_counts().filter(like='oppgave2g', axis=0)\n",
    "\n",
    "#Group the data into values that contain \"oppgave2h\"\n",
    "oppgave2h = data.stack().value_counts().filter(like='oppgave2h', axis=0)\n",
    "\n",
    "#Group the data into values that contain \"oppgave2i\"\n",
    "oppgave2i = data.stack().value_counts().filter(like='oppgave2i', axis=0)\n",
    "\n",
    "#Group the data into values that contain \"oppgave3_1\"\n",
    "oppgave3_1 = data.stack().value_counts().filter(like='oppgave3_1', axis=0)\n",
    "\n",
    "#Group the data into values that contain \"oppgave3_2\"\n",
    "oppgave3_2 = data.stack().value_counts().filter(like='oppgave3_2', axis=0)\n",
    "\n",
    "#Group the data into values that contain \"oppgave3_3\"\n",
    "oppgave3_3 = data.stack().value_counts().filter(like='oppgave3_3', axis=0)\n",
    "\n",
    "#Group the data into values that contain \"oppgave3_4\"\n",
    "oppgave3_4 = data.stack().value_counts().filter(like='oppgave3_4', axis=0)\n",
    "\n",
    "#Group the data into values that contain \"oppgave3_5\"\n",
    "oppgave3_5 = data.stack().value_counts().filter(like='oppgave3_5', axis=0)\n",
    "\n",
    "#Group the data into values that contain \"oppgave3_6\"\n",
    "oppgave3_6 = data.stack().value_counts().filter(like='oppgave3_6', axis=0)\n",
    "\n",
    "#Print the counts of the grouped data\n",
    "print(\"Oppgave 2a: \", oppgave2a.sum())\n",
    "print(\"Oppgave 2b: \", oppgave2b.sum())\n",
    "print(\"Oppgave 2c: \", oppgave2c.sum())\n",
    "print(\"Oppgave 2g: \", oppgave2g.sum())\n",
    "print(\"Oppgave 2h: \", oppgave2h.sum())\n",
    "print(\"Oppgave 2i: \", oppgave2i.sum())\n",
    "print(\"Oppgave 3.1: \", oppgave3_1.sum())\n",
    "print(\"Oppgave 3.2: \", oppgave3_2.sum())\n",
    "print(\"Oppgave 3.3: \", oppgave3_3.sum())\n",
    "print(\"Oppgave 3.4: \", oppgave3_4.sum())\n",
    "print(\"Oppgave 3.5: \", oppgave3_5.sum())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Oppgave 1:  372\n",
      "Oppgave 2:  782\n",
      "Oppgave 3:  353\n"
     ]
    }
   ],
   "source": [
    "# Load csv file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H22/errorH22morgen.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Group the data into values that contain \"oppgave1\"\n",
    "oppgave1 = data.stack().value_counts().filter(like='oppgave1', axis=0)\n",
    "\n",
    "#Group the data into values that contain \"oppgave2\"\n",
    "oppgave2 = data.stack().value_counts().filter(like='oppgave2', axis=0)\n",
    "\n",
    "#Group the data into values that contain \"oppgave3\"\n",
    "oppgave3 = data.stack().value_counts().filter(like='oppgave3', axis=0)\n",
    "\n",
    "#Print the counts of the grouped data\n",
    "print(\"Oppgave 1: \", oppgave1.sum())\n",
    "print(\"Oppgave 2: \", oppgave2.sum())\n",
    "print(\"Oppgave 3: \", oppgave3.sum())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Oppgave 1:  51\n",
      "Oppgave 2:  29\n",
      "Oppgave 3:  96\n"
     ]
    }
   ],
   "source": [
    "# Load csv file into a pandas DataFrame\n",
    "data = pd.read_csv(\"H22Kveld/errorH22kveld.csv\", on_bad_lines='skip')\n",
    "\n",
    "#Group the data into values that contain \"oppgave1\"\n",
    "oppgave1 = data.stack().value_counts().filter(like='oppgave1', axis=0)\n",
    "\n",
    "#Group the data into values that contain \"oppgave2\"\n",
    "oppgave2 = data.stack().value_counts().filter(like='oppgave2', axis=0)\n",
    "\n",
    "#Group the data into values that contain \"oppgave3\"\n",
    "oppgave3 = data.stack().value_counts().filter(like='oppgave3', axis=0)\n",
    "\n",
    "#Print the counts of the grouped data\n",
    "print(\"Oppgave 1: \", oppgave1.sum())\n",
    "print(\"Oppgave 2: \", oppgave2.sum())\n",
    "print(\"Oppgave 3: \", oppgave3.sum())"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
